app/models/user.rb

```diff
@@ -4,6 +4,17 @@ class User < ApplicationRecord
 
   has_many :articles, foreign_key: 'author_id', dependent: :destroy
   has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  has_many :liked_articles, through: :likes, source: :article
 
   validates :username, presence: true, length: { minimum: 2, maximum: 90 }
+
+  def like(article)
+    like = likes.build(article: article)
+    likes << like
+  end
+
+  def unlike(article)
+    likes.find_by(article: article).try(:destroy)
```

Comment:
and this.

Also, we have an idea that if you need to use 'try' somewhere - it means that you need to re-create that code and dont use 'try' :)